### PR TITLE
Added `--gpu-index` to `forwardable_cli_arguments`

### PR DESCRIPTION
**Pull request by Bestest-Coder**
when the gpu index is specified through the CLI, that setting will be inherited by both the editor (if started through project manager) and instances of the game started through the editor

I feel this should be default behavior, as if a user specifies a GPU to be used it should be shared from project manager to editor, and editor to debug instance. This can still be overwritten at runtime with the project setting `editor/run/main_run_args`

### CODE REVIEW

**Review by Calinou — APPROVED** [review]
Tested locally (using `--rendering-method forward_plus --gpu-index 1` on Linux), it works.

You can test this on a system with a single GPU on Linux by having llvmpipe/Lavapipe installed, then open a project that uses Forward+ or Mobile from the project manager. This works too if the project manager is started with OpenGL, which is the default unless you specify `--rendering-method forward_plus`.

File: main/main.cpp [hunk 1]
```diff
@@ -856,6 +856,15 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 				forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(I->next()->get());
 			}
 		}
+		// if gpu is specified, both editor and debug instances started from editor will inherit
```
Comment (akien-mga) [inline main/main.cpp:859]:
```suggestion
		// If gpu is specified, both editor and debug instances started from editor will inherit.
```

**Review by akien-mga — APPROVED** [review]
Seems ok, though I find all this increasingly hacky. Only some options get forwarded, and they get forwarded to different contexts. It's pretty unpredictable without reading the source code.

There may be use cases such as running the editor on an iGPU to save battery during development, while running the game on dGPU because it requires all the performance. This change would prevent doing this as the setting will propagate to both editor and game.

But I guess it's not a big deal for such advanced options.

**Comment by Bestest-Coder** [comment]
You would still be able to do so, as specifying `--gpu-index` again in the project setting `editor/run/main_run_args` will overwrite it and cause the debug instance to be run on the newly specified GPU

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by Bestest-Coder** [comment]
Anything else need to be done?

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**